app/src/main/java/com/nexenio/bleindoorpositioningdemo/ui/beaconview/map/BeaconMapBackground.java

```diff
@@ -0,0 +1,214 @@
+    }
+
+    public Location getLocation(Point point) {
+        return getLocation(point, topLeftLocation, topLeftPoint, metersPerPixel, bearing);
+    }
+
+    public Point getPoint(Location location) {
+        return getPoint(location, topLeftLocation, topLeftPoint, metersPerPixel, bearing);
+    }
+
+    public static float getMetersPerPixel(Location firstReferenceLocation, Point firstReferencePoint, Location secondReferenceLocation, Point secondReferencePoint) {
+        double distanceInPixels = getPixelDistance(firstReferencePoint, secondReferencePoint);
+        double distanceInMeters = firstReferenceLocation.getDistanceTo(secondReferenceLocation);
+        return (float) (distanceInMeters / distanceInPixels);
+    }
+
+    public static double getBearing(Location firstReferenceLocation, Point firstReferencePoint, Location secondReferenceLocation, Point secondReferencePoint) {
+        double locationBearing = getBearing(firstReferenceLocation, secondReferenceLocation);
+        double pointBearing = getBearing(firstReferencePoint, secondReferencePoint);
+        return ((locationBearing - pointBearing) + 360) % 360;
+    }
+
+    public static double getBearing(Point firstPoint, Point secondPoint) {
+        double angle = Math.atan2(secondPoint.y - firstPoint.y, secondPoint.x - firstPoint.x) * 180 / Math.PI;
+        return (angle + 90) % 360;
+    }
+
+    public static double getBearing(Location firstLocation, Location secondLocation) {
+        return firstLocation.getAngleTo(secondLocation);
+    }
+
+    public static double getPixelDistance(Point firstPoint, Point secondPoint) {
+        int deltaX = Math.abs(secondPoint.x - firstPoint.x);
+        int deltaY = Math.abs(secondPoint.y - firstPoint.y);
+        return Math.sqrt((deltaX * deltaX) + (deltaY * deltaY));
+    }
+
+    public static Location getLocation(Point point, Location referenceLocation, Point referencePoint, double metersPerPixel, double backgroundBearing) {
+        double distance = metersPerPixel * getPixelDistance(referencePoint, point);
+        double bearing = (getBearing(referencePoint, point) + backgroundBearing + 360) % 360;
+        return referenceLocation.getShiftedLocation(distance, bearing);
+    }
+
+    public static Point getPoint(Location location, Location referenceLocation, Point referencePoint, double metersPerPixel, double backgroundBearing) {
+        double distanceInMeters = location.getDistanceTo(referenceLocation);
+        double distanceInPixels = distanceInMeters / metersPerPixel;
+        double locationBearing = getBearing(referenceLocation, location);
+        double bearing = (locationBearing - backgroundBearing + 360) % 360;
+        return getShiftedPoint(referencePoint, distanceInPixels, bearing);
+    }
+
+    public static Point getShiftedPoint(Point referencePoint, double distanceInPixels, double bearing) {
+        double angleInRadians = Math.toRadians(bearing + 90);
```

Comment:
Where does the `+90` come from?

Reply:
I guess that relates to the starting point of the circle (0°) being on the x axis (east, so 90° shifted from north), but we use bearing, where 0° should be north. It's also used in `getBearing`:

```java
public static double getBearing(@NonNull Point firstPoint, @NonNull Point secondPoint) {
    double angle = Math.atan2(secondPoint.y - firstPoint.y, secondPoint.x - firstPoint.x) * 180 / Math.PI;
    return (angle + 90) % 360;
}
```